diff --git a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.c b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.c
index 8b4c721..91d4000 100644
--- a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.c
+++ b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.c
@@ -24,6 +24,10 @@
 #include <math.h>
 #include "inv_mpu.h"
 
+#ifdef ESP_PLATFORM
+#include "../../../../../../hal/include/mpu_idf_esp32.h"
+#endif
+
 /* The following functions must be defined for this platform:
  * i2c_write(unsigned char slave_addr, unsigned char reg_addr,
  *      unsigned char length, unsigned char const *data)
@@ -101,6 +105,13 @@ static inline int reg_int_cb(struct int_param_s *int_param)
 /* UC3 is a 32-bit processor, so abs and labs are equivalent. */
 #define labs        abs
 #define fabs(x)     (((x)>0)?(x):-(x))
+
+#elif defined CONFIG_IDF_TARGET_ESP32
+#define delay_ms    esp32_delay_ms
+#define get_ms      esp32_get_clock_ms
+#define log_i       MPL_LOGI
+#define log_e       MPL_LOGE
+
 #else
 #error  Gyro driver is missing the system layer implementations.
 #endif
diff --git a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.h b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.h
index 01a47d1..6c27fd2 100644
--- a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.h
+++ b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.h
@@ -34,6 +34,10 @@ struct int_param_s {
     unsigned short pin;
     unsigned char lp_exit;
     unsigned char active_low;
+#elif defined ESP_PLATFORM
+    unsigned int interrupt_pin;
+    void (*cb)(void*);
+    void * arg;
 #elif defined EMPL_TARGET_UC3L0
     unsigned long pin;
     void (*cb)(volatile void*);
diff --git a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu_dmp_motion_driver.c b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu_dmp_motion_driver.c
index 02403b0..077563c 100644
--- a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu_dmp_motion_driver.c
+++ b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu_dmp_motion_driver.c
@@ -24,6 +24,10 @@
 #include "dmpKey.h"
 #include "dmpmap.h"
 
+#ifdef ESP_PLATFORM
+#include "../../../../../../hal/include/mpu_idf_esp32.h"
+#endif
+
 /* The following functions must be defined for this platform:
  * i2c_write(unsigned char slave_addr, unsigned char reg_addr,
  *      unsigned char length, unsigned char const *data)
@@ -62,6 +66,12 @@
 #define log_i       MPL_LOGI
 #define log_e       MPL_LOGE
 
+#elif defined CONFIG_IDF_TARGET_ESP32
+#define delay_ms    esp32_delay_ms
+#define get_ms      esp32_get_clock_ms
+#define log_i       MPL_LOGI
+#define log_e       MPL_LOGE
+
 #else
 #error  Gyro driver is missing the system layer implementations.
 #endif
diff --git a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mlinclude.h b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mlinclude.h
index 9725199..e0becf3 100644
--- a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mlinclude.h
+++ b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mlinclude.h
@@ -6,6 +6,8 @@
 #ifndef INV_INCLUDE_H__
 #define INV_INCLUDE_H__
 
+#pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+
 #define INVENSENSE_FUNC_START  typedef int invensensePutFunctionCallsHere
 
 #ifdef COVERAGE
diff --git a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mltypes.h b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mltypes.h
index c4fd63f..a2b5e58 100644
--- a/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mltypes.h
+++ b/components/mpu/motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mltypes.h
@@ -129,7 +129,7 @@ typedef uint64_t  __u64;
 #ifndef __cplusplus
 #ifndef __KERNEL__
 #ifndef EMPL_TARGET_UC3L0
-typedef int_fast8_t bool;
+//typedef int_fast8_t bool;
 #endif
 
 #ifndef false
